### ISSUE TITLE
Task: Remove publicDashboards FF

### ISSUE THREAD

**Issue by juanicabanas** [grafana/grafana#78050]

**Comment by AgnesToulet** [comment]
@juanicabanas @natellium @thanos-karachalios I started to work on this and I think we should replace it with a configuration field to allow users to disable the feature. As this has security implications, I wonder if we should do this in a two-step process:
1. Add configuration field, keep the feature toggle but add deprecation warnings if this is used to disable the feature to tell users to use the configuration field instead.
2. Entirely remove the feature toggle. 

WDYT?

This is a bit more work and this won't impact a lot of customers. Currently, there are 68 instances with the feature disabled on Cloud VS 1515 enabled: https://ops.grafana-ops.net/d/afqrz7j7z/public-dashboards?orgId=1

**Comment by natellium** [comment]
No strong opinion from my side. Your plan makes sense to me, we may wanna set a date by which we'll fully remove the toggle?

**Comment by juanicabanas** [comment]
@AgnesToulet  I agree with the steps!

**Comment by AgnesToulet** [comment]
@natellium Yes, we should set a date and use it in the deprecation warning. I'm thinking either next minor (10.3.0, planned on 2024-01-23) or next major (11.0.0, planned on 2024-04-16).
